VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/SwitchIsMissingDefaultLabel/SwitchIsMissingDefaultLabelCodeFix.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.SwitchIsMissingDefaultLabel
+{
+    [ExportCodeFixProvider(DiagnosticId.SwitchIsMissingDefaultLabel + "CF", LanguageNames.CSharp), Shared]
+    internal class SwitchIsMissingDefaultLabelCodeFix : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(SwitchIsMissingDefaultLabelAnalyzer.Rule.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var statement = root.FindNode(diagnosticSpan);
+                context.RegisterCodeFix(
+                    CodeAction.Create(VSDiagnosticsResources.SwitchIsMissingDefaultSectionCodeFixTitle,
+                        x => AddDefaultCaseAsync(context.Document, (CompilationUnitSyntax) root, statement),
+                        SwitchIsMissingDefaultLabelAnalyzer.Rule.Id), diagnostic);
+        }
+
+        private async Task<Solution> AddDefaultCaseAsync(Document document, CompilationUnitSyntax root, SyntaxNode statement)
+        {
+            var switchBlock = (SwitchStatementSyntax)statement.Parent;
```

Comment:
This will cause an exception when you have a parenthesized switch like this:

```csharp
	var x = 5;

	switch ((x))
	{
		case 5: 
		case 6:
		default:
			break;
	}
```

Add this as a test. You can probably just look through the ancestors to find the first of `SyntaxKind.SwitchStatement`.